src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -3647,6 +3647,114 @@ public void SyntaxNodeContains()
             Assert.True(firstParens.Contains(e));
         }
 
+        [Fact, WorkItem(54239, "https://github.com/dotnet/roslyn/issues/54239")]
+        public void TestWithAsyncKeyword_AnonymousMethodExpressionSyntax_AddAsync()
+        {
+            var text = "static delegate(int i) { }";
+            var expression = (AnonymousMethodExpressionSyntax)SyntaxFactory.ParseExpression(text);
+            var withAsync = expression.WithAsyncKeyword(SyntaxFactory.Token(SyntaxKind.AsyncKeyword).WithTrailingTrivia(SyntaxFactory.Space)).ToString();
+            Assert.Equal("static async delegate(int i) { }", withAsync);
```

Comment:
>Assert.Equal("static async delegate(int i) { }", withAsync); [](http://example.com/codeflow?start=12&length=60)

Is this what we normally do for tests like this, i.e checking the text rather than the shape of the tree that the API is supposed to create?

Reply:
@AlekseyTs I think there is an inconsistency here among the existing tests. For example:

https://github.com/dotnet/roslyn/blob/1c7050e4cb294031f8b3030c6b090db159105962/src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs#L49-L57

IMO it's okay to simply check the string in this case. We just want to assert that static doesn't get removed, which the current test does.

But I'm open to change the test here if you prefer some other form.